File: Backend/src/main/java/kauan/projetcts/TechStore/Domain/Produto.java
package kauan.projetcts.TechStore.Domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import net.minidev.json.annotate.JsonIgnore;

@Getter
@Setter
@Entity
public class Produto {
    @Id
    @GeneratedValue
    private int id;
    private String produtoNome;
    private String produtoDescricao;
    private int produtoValor;
    private double produtoReview;
    private int reviewsCount;
    private String produtoImage;
    private ProdutoCategoria produtoCategoria;


}

File: frontend/app/types/produto.ts
export enum ProdutoCategoria {
  COMPONENTES = "COMPONENTES",
  CONTROLES = "CONTROLES",
  MONITORES = "MONITORES",
  PROCESSADORES = "PROCESSADORES",
  NOTEBOOKS = "NOTEBOOKS",
  PERIFERICOS = "PERIFERICOS",
  ACESSORIOS = "ACESSORIOS",
}
export interface Produto {
  id: number;
  produtoNome: string;
  produtoDescricao: string;
  produtoValor: number;
  produtoReview: number;
  reviewsCount: number;
  produtoImage: string;
  produtoCategoria: ProdutoCategoria;
}
